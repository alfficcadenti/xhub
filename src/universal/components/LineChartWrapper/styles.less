@import '../../mixins.less';

.line-chart-wrapper {
    .custom-tooltip-container {
        color: @color-text-inverse;
        background: @color-bg-inverse;
        opacity: .8;
        min-width: 260px;
        padding: @spacing-1x;

        .dot-date {
            font-weight: 800;
        }

        .dot-title {
            font-style: italic;
        }
    }

    .help-text .Tooltip .Tooltip__inner {
        font-weight: 400;
    }
}
