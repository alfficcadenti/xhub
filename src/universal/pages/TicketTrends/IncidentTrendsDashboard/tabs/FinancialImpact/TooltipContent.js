import React from 'react';

const htmlDecode = (input) => {
    const doc = new DOMParser().parseFromString(input, 'text/html');
    return doc.documentElement.textContent;
};

const TooltipContent = ({link, lostRevenue}) => (
    <div key={link} className="incident-wrapper">
        <div
            className="incident-number"
            dangerouslySetInnerHTML={{ //eslint-disable-line
                __html: htmlDecode(link)
            }}
        />
        <div className="incident-financial-impact">
            {lostRevenue.toLocaleString('en-US', {style: 'currency', currency: 'USD'})}
        </div>
    </div>
);

export default TooltipContent;
