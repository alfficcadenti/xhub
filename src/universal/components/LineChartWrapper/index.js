import React, {useState} from 'react';
import {
    LineChart, Line, XAxis, YAxis, CartesianGrid, Tooltip, Legend, ResponsiveContainer, ReferenceArea
} from 'recharts';
import HelpText from '../../components/HelpText/HelpText';
import {CHART_COLORS} from '../../constants';
import './styles.less';

// eslint-disable-next-line complexity
const LineChartWrapper = ({
    title,
    helpText,
    data = [],
    keys = [],
    onDotClick,
    tooltipData,
    renderTooltipContent,
    enableLineHiding = false,
    refAreaLeft,
    refAreaRight,
    onMouseDown = () => {},
    onMouseMove = () => {},
    onMouseUp = () => {},
    showDot = true
}) => {
    const [selectedLine, setSelectedLine] = useState();
    const [hiddenKeys, setHiddenKeys] = useState([]);

    const yAxisId = `${title}-yAxis`;

    const CustomTooltip = ({label, active, payload}) => {
        if (!active || !selectedLine) {
            return null;
        }
        for (const line of payload) {
            if (line.dataKey === selectedLine) {
                return (
                    <div className="custom-tooltip-container">
                        <div className="dot-date">{label}</div>
                        <div className="dot-title">{line.name}</div>
                        {tooltipData[label][line.name].map(renderTooltipContent)}
                    </div>
                );
            }
        }
        return null;
    };

    const handleLegendClick = (e) => {
        if (e && e.dataKey) {
            const nextHiddenKeys = [...hiddenKeys];
            const foundIdx = hiddenKeys.findIndex((h) => h === e.dataKey);
            if (foundIdx > -1) {
                nextHiddenKeys.splice(foundIdx, 1);
            } else {
                nextHiddenKeys.push(e.dataKey);
            }
            setHiddenKeys(nextHiddenKeys);
        }
    };

    const renderLine = (line, idx) => (
        <Line
            key={`line-${title}-${line}-${idx}`}
            type="monotone"
            dataKey={line}
            stroke={CHART_COLORS[idx]}
            activeDot={{
                onMouseOver: () => setSelectedLine(line),
                onClick: onDotClick || (() => {})
            }}
            hide={hiddenKeys.includes(line)}
            yAxisId={yAxisId}
            dot={showDot}
        />
    );

    const renderTitle = () => (
        <h3>
            {title}
            {helpText && <HelpText className="help-text" text={helpText} placement="bottom" />}
        </h3>
    );

    return (
        <div className="line-chart-wrapper">
            {title && renderTitle()}
            <ResponsiveContainer width="100%" height={300}>
                <LineChart
                    data={data}
                    margin={{top: 5, right: 30, left: 20, bottom: 5}}
                    cursor={onDotClick ? 'pointer' : ''}
                    onMouseDown={onMouseDown}
                    onMouseMove={onMouseMove}
                    onMouseUp={onMouseUp}
                >
                    <CartesianGrid strokeDasharray="3 3" />
                    <XAxis dataKey="name" />
                    <YAxis yAxisId={yAxisId} allowDecimals={false} type="number" />
                    <Legend onClick={handleLegendClick} cursor={enableLineHiding ? 'pointer' : ''} />
                    {!tooltipData || !renderTooltipContent
                        ? <Tooltip />
                        : <Tooltip content={<CustomTooltip />} />
                    }
                    {keys.map(renderLine)}
                    {
                        (refAreaLeft && refAreaRight)
                            ? <ReferenceArea yAxisId={yAxisId} x1={refAreaLeft} x2={refAreaRight} strokeOpacity={0.3} />
                            : null
                    }
                </LineChart>
            </ResponsiveContainer>
        </div>
    );
};

export default React.memo(LineChartWrapper);
