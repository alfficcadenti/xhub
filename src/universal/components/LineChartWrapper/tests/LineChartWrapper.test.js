import React from 'react';
import {expect} from 'chai';
import {shallow, mount} from 'enzyme';
import LineChartWrapper from '../';
import TooltipContent from '../../../pages/TicketTrends/IncidentTrendsDashboard/tabs/FinancialImpact/TooltipContent';


describe('<LineChartWrapper />', () => {
    let wrapper;

    it('renders correctly', () => {
        wrapper = shallow(<LineChartWrapper data={[{name: 'name', value: 'value'}]} keys={['value']} />);
        expect(wrapper).to.have.length(1);
        expect(wrapper.find('h3').children()).to.have.length(0);
    });

    it('displays a tooltip when active', () => {
        const tooltipData = {
            '2021-06-20': {
                'Egencia': [],
                'Expedia': [
                    {
                        link: '<a href=\"https://expedia.service-now.com/go.do?id=INC5661034\" target=\"_blank\" data-reactroot=\"\">INC5661034</a>',
                        lostRevenue: 843219
                    },
                    {
                        link: '<a href=\"https://expedia.service-now.com/go.do?id=INC5653653\" target=\"_blank\" data-reactroot=\"\">INC5653653</a>',
                        lostRevenue: 62142
                    },
                    {
                        link: '<a href=\"https://expedia.service-now.com/go.do?id=EPM-1002\" target=\"_blank\" data-reactroot=\"\">EPM-1002</a>',
                        lostRevenue: 0
                    },
                    {
                        link: '<a href=\"https://expedia.service-now.com/go.do?id=EPM-1003\" target=\"_blank\" data-reactroot=\"\">EPM-1003</a>',
                        lostRevenue: 0
                    },
                    {
                        link: '<a href=\"https://expedia.service-now.com/go.do?id=EPM-1005\" target=\"_blank\" data-reactroot=\"\">EPM-1005</a>',
                        lostRevenue: 0
                    }],
                'Expedia Group': [],
                'Expedia Partner Solutions': [],
                'Hotels.com': [{
                    link: '<a href=\"https://expedia.service-now.com/go.do?id=INC5652612\" target=\"_blank\" data-reactroot=\"\">INC5652612</a>',
                    lostRevenue: 367080
                }],
                Vrbo: [{
                    link: '<a href=\"https://expedia.service-now.com/go.do?id=INC5660823\" target=\"_blank\" data-reactroot=\"\">INC5660823</a>',
                    lostRevenue: 972046
                }],
            }
        };

        const onMouseDown = () => {};
        const onMouseMove = () => {};
        const onMouseUp = () => {};
        const onDotClick = () => {};
        const payload = [{
            color: '#3366cc',
            dataKey: 'Expedia',
            fill: '#fff',
            formatter: null,
            name: 'Expedia',
            payload: {
                'Egencia': 0,
                'Expedia': 905361,
                'Expedia Group': 0,
                'Expedia Partner Solutions': 0,
                'Hotels.com': 367080,
                'Vrbo': 972046,
                name: '2021-06-20',
            },
            stroke: '#3366cc',
            strokeWidth: 1,
            type: null,
            unit: null,
            value: 0,
        }];

        wrapper = mount(<LineChartWrapper
            width={700}
            height={300}
            title="title" data={[{name: 'name', value: 'value'}]}
            keys={['value']}
            active
            selectedLine="Expedia"
            tooltipData={tooltipData}
            renderTooltipContent={TooltipContent}
            margin={{top: 5, right: 30, left: 20, bottom: 5}}
            cursor={onDotClick ? 'pointer' : ''}
            onMouseDown={onMouseDown}
            onMouseMove={onMouseMove}
            onMouseUp={onMouseUp}
            label="2021-06-20"
            payload={payload}
        />);

        expect(wrapper.find('.recharts-tooltip-wrapper')).to.have.length(1);
        expect(wrapper.find('.recharts-legend-wrapper')).to.have.length(1);
        wrapper.find('.recharts-legend-wrapper li').simulate('click');
        expect(wrapper.find('.legend-item-0.inactive')).to.have.length(1);
    });
});
