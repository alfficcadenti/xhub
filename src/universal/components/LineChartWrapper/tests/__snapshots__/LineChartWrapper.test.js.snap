// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LineChartWrapper /> renders correctly and enable/disable the line when click on the legend 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="line-chart-wrapper"
        title="title test"
      >
        <h3>
          title test
        </h3>
        <div
          class="recharts-wrapper"
          style="position: relative; cursor: default; width: 700px; height: 300px;"
        >
          <svg
            class="recharts-surface"
            cursor=""
            height="300"
            version="1.1"
            viewBox="0 0 700 300"
            width="700"
          >
            <defs>
              <clippath
                id="recharts2-clip"
              >
                <rect
                  height="260"
                  width="590"
                  x="80"
                  y="5"
                />
              </clippath>
            </defs>
            <g
              class="recharts-cartesian-grid"
            >
              <g
                class="recharts-cartesian-grid-horizontal"
              >
                <line
                  fill="none"
                  height="260"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="590"
                  x="80"
                  x1="80"
                  x2="670"
                  y="5"
                  y1="5"
                  y2="5"
                />
                <line
                  fill="none"
                  height="260"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="590"
                  x="80"
                  x1="80"
                  x2="670"
                  y="5"
                  y1="265"
                  y2="265"
                />
              </g>
              <g
                class="recharts-cartesian-grid-vertical"
              >
                <line
                  fill="none"
                  height="260"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="590"
                  x="80"
                  x1="375"
                  x2="375"
                  y="5"
                  y1="5"
                  y2="265"
                />
                <line
                  fill="none"
                  height="260"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="590"
                  x="80"
                  x1="80"
                  x2="80"
                  y="5"
                  y1="5"
                  y2="265"
                />
                <line
                  fill="none"
                  height="260"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="590"
                  x="80"
                  x1="670"
                  x2="670"
                  y="5"
                  y1="5"
                  y2="265"
                />
              </g>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
            >
              <line
                class="recharts-cartesian-axis-line"
                fill="none"
                height="30"
                stroke="#666"
                width="590"
                x="80"
                x1="80"
                x2="670"
                y="265"
                y1="265"
                y2="265"
              />
              <g
                class="recharts-cartesian-axis-ticks"
              >
                <g
                  class="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    class="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height="30"
                    stroke="#666"
                    width="590"
                    x="80"
                    x1="375"
                    x2="375"
                    y="265"
                    y1="271"
                    y2="265"
                  />
                  <text
                    class="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height="30"
                    stroke="none"
                    text-anchor="middle"
                    width="590"
                    x="375"
                    y="273"
                  >
                    <tspan
                      dy="0.71em"
                      x="375"
                    >
                      name
                    </tspan>
                  </text>
                </g>
              </g>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
            >
              <line
                class="recharts-cartesian-axis-line"
                fill="none"
                height="260"
                stroke="#666"
                width="60"
                x="20"
                x1="80"
                x2="80"
                y="5"
                y1="5"
                y2="265"
              />
              <g
                class="recharts-cartesian-axis-ticks"
              />
            </g>
            <g
              class="recharts-layer recharts-line"
            >
              <g
                class="recharts-layer recharts-line-dots"
              />
            </g>
          </svg>
          <div
            class="recharts-legend-wrapper"
            style="position: absolute; width: 650px; height: auto; left: 20px; bottom: 5px;"
          >
            <ul
              class="recharts-default-legend"
              style="padding: 0px; margin: 0px; text-align: center;"
            >
              <li
                class="recharts-legend-item legend-item-0"
                style="display: inline-block; margin-right: 10px;"
              >
                <svg
                  class="recharts-surface"
                  height="14"
                  style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                  version="1.1"
                  viewBox="0 0 32 32"
                  width="14"
                >
                  <path
                    class="recharts-legend-icon"
                    d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
                    fill="none"
                    stroke="#3366cc"
                    stroke-width="4"
                  />
                </svg>
                <span
                  class="recharts-legend-item-text"
                >
                  value
                </span>
              </li>
            </ul>
          </div>
          <div
            class="recharts-tooltip-wrapper"
            style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px;"
          >
            <div
              class="recharts-default-tooltip"
              style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
            >
              <p
                class="recharts-tooltip-label"
                style="margin: 0px;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <span
      id="recharts_measurement_span"
      style="position: absolute; top: -20000px; left: 0px; padding: 0px; margin: 0px; white-space: pre;"
    >
      Â 
    </span>
  </body>,
  "container": <div>
    <div
      class="line-chart-wrapper"
      title="title test"
    >
      <h3>
        title test
      </h3>
      <div
        class="recharts-wrapper"
        style="position: relative; cursor: default; width: 700px; height: 300px;"
      >
        <svg
          class="recharts-surface"
          cursor=""
          height="300"
          version="1.1"
          viewBox="0 0 700 300"
          width="700"
        >
          <defs>
            <clippath
              id="recharts2-clip"
            >
              <rect
                height="260"
                width="590"
                x="80"
                y="5"
              />
            </clippath>
          </defs>
          <g
            class="recharts-cartesian-grid"
          >
            <g
              class="recharts-cartesian-grid-horizontal"
            >
              <line
                fill="none"
                height="260"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="590"
                x="80"
                x1="80"
                x2="670"
                y="5"
                y1="5"
                y2="5"
              />
              <line
                fill="none"
                height="260"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="590"
                x="80"
                x1="80"
                x2="670"
                y="5"
                y1="265"
                y2="265"
              />
            </g>
            <g
              class="recharts-cartesian-grid-vertical"
            >
              <line
                fill="none"
                height="260"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="590"
                x="80"
                x1="375"
                x2="375"
                y="5"
                y1="5"
                y2="265"
              />
              <line
                fill="none"
                height="260"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="590"
                x="80"
                x1="80"
                x2="80"
                y="5"
                y1="5"
                y2="265"
              />
              <line
                fill="none"
                height="260"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="590"
                x="80"
                x1="670"
                x2="670"
                y="5"
                y1="5"
                y2="265"
              />
            </g>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
          >
            <line
              class="recharts-cartesian-axis-line"
              fill="none"
              height="30"
              stroke="#666"
              width="590"
              x="80"
              x1="80"
              x2="670"
              y="265"
              y1="265"
              y2="265"
            />
            <g
              class="recharts-cartesian-axis-ticks"
            >
              <g
                class="recharts-layer recharts-cartesian-axis-tick"
              >
                <line
                  class="recharts-cartesian-axis-tick-line"
                  fill="none"
                  height="30"
                  stroke="#666"
                  width="590"
                  x="80"
                  x1="375"
                  x2="375"
                  y="265"
                  y1="271"
                  y2="265"
                />
                <text
                  class="recharts-text recharts-cartesian-axis-tick-value"
                  fill="#666"
                  height="30"
                  stroke="none"
                  text-anchor="middle"
                  width="590"
                  x="375"
                  y="273"
                >
                  <tspan
                    dy="0.71em"
                    x="375"
                  >
                    name
                  </tspan>
                </text>
              </g>
            </g>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
          >
            <line
              class="recharts-cartesian-axis-line"
              fill="none"
              height="260"
              stroke="#666"
              width="60"
              x="20"
              x1="80"
              x2="80"
              y="5"
              y1="5"
              y2="265"
            />
            <g
              class="recharts-cartesian-axis-ticks"
            />
          </g>
          <g
            class="recharts-layer recharts-line"
          >
            <g
              class="recharts-layer recharts-line-dots"
            />
          </g>
        </svg>
        <div
          class="recharts-legend-wrapper"
          style="position: absolute; width: 650px; height: auto; left: 20px; bottom: 5px;"
        >
          <ul
            class="recharts-default-legend"
            style="padding: 0px; margin: 0px; text-align: center;"
          >
            <li
              class="recharts-legend-item legend-item-0"
              style="display: inline-block; margin-right: 10px;"
            >
              <svg
                class="recharts-surface"
                height="14"
                style="display: inline-block; vertical-align: middle; margin-right: 4px;"
                version="1.1"
                viewBox="0 0 32 32"
                width="14"
              >
                <path
                  class="recharts-legend-icon"
                  d="M0,16h10.666666666666666
            A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
            H32M21.333333333333332,16
            A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
                  fill="none"
                  stroke="#3366cc"
                  stroke-width="4"
                />
              </svg>
              <span
                class="recharts-legend-item-text"
              >
                value
              </span>
            </li>
          </ul>
        </div>
        <div
          class="recharts-tooltip-wrapper"
          style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px;"
        >
          <div
            class="recharts-default-tooltip"
            style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
          >
            <p
              class="recharts-tooltip-label"
              style="margin: 0px;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<LineChartWrapper /> renders correctly when data and keys are not passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <span
      id="recharts_measurement_span"
      style="position: absolute; top: -20000px; left: 0px; padding: 0px; margin: 0px; white-space: pre;"
    >
      Â 
    </span>
    <div>
      <div
        class="line-chart-wrapper"
        title="title test"
      >
        <h3>
          title test
        </h3>
        <div
          class="recharts-wrapper"
          style="position: relative; cursor: default; width: 700px; height: 300px;"
        >
          <svg
            class="recharts-surface"
            cursor=""
            height="300"
            version="1.1"
            viewBox="0 0 700 300"
            width="700"
          >
            <defs>
              <clippath
                id="recharts5-clip"
              >
                <rect
                  height="260"
                  width="590"
                  x="80"
                  y="5"
                />
              </clippath>
            </defs>
            <g
              class="recharts-cartesian-grid"
            >
              <g
                class="recharts-cartesian-grid-horizontal"
              >
                <line
                  fill="none"
                  height="260"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="590"
                  x="80"
                  x1="80"
                  x2="670"
                  y="5"
                  y1="5"
                  y2="5"
                />
                <line
                  fill="none"
                  height="260"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="590"
                  x="80"
                  x1="80"
                  x2="670"
                  y="5"
                  y1="265"
                  y2="265"
                />
              </g>
              <g
                class="recharts-cartesian-grid-vertical"
              >
                <line
                  fill="none"
                  height="260"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="590"
                  x="80"
                  x1="80"
                  x2="80"
                  y="5"
                  y1="5"
                  y2="265"
                />
                <line
                  fill="none"
                  height="260"
                  stroke="#ccc"
                  stroke-dasharray="3 3"
                  width="590"
                  x="80"
                  x1="670"
                  x2="670"
                  y="5"
                  y1="5"
                  y2="265"
                />
              </g>
            </g>
            <g
              class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
            >
              <line
                class="recharts-cartesian-axis-line"
                fill="none"
                height="260"
                stroke="#666"
                width="60"
                x="20"
                x1="80"
                x2="80"
                y="5"
                y1="5"
                y2="265"
              />
              <g
                class="recharts-cartesian-axis-ticks"
              />
            </g>
          </svg>
          <div
            class="recharts-legend-wrapper"
            style="position: absolute; width: 650px; height: auto; left: 20px; bottom: 5px;"
          />
          <div
            class="recharts-tooltip-wrapper"
            style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px;"
          >
            <div
              class="recharts-default-tooltip"
              style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
            >
              <p
                class="recharts-tooltip-label"
                style="margin: 0px;"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="line-chart-wrapper"
      title="title test"
    >
      <h3>
        title test
      </h3>
      <div
        class="recharts-wrapper"
        style="position: relative; cursor: default; width: 700px; height: 300px;"
      >
        <svg
          class="recharts-surface"
          cursor=""
          height="300"
          version="1.1"
          viewBox="0 0 700 300"
          width="700"
        >
          <defs>
            <clippath
              id="recharts5-clip"
            >
              <rect
                height="260"
                width="590"
                x="80"
                y="5"
              />
            </clippath>
          </defs>
          <g
            class="recharts-cartesian-grid"
          >
            <g
              class="recharts-cartesian-grid-horizontal"
            >
              <line
                fill="none"
                height="260"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="590"
                x="80"
                x1="80"
                x2="670"
                y="5"
                y1="5"
                y2="5"
              />
              <line
                fill="none"
                height="260"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="590"
                x="80"
                x1="80"
                x2="670"
                y="5"
                y1="265"
                y2="265"
              />
            </g>
            <g
              class="recharts-cartesian-grid-vertical"
            >
              <line
                fill="none"
                height="260"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="590"
                x="80"
                x1="80"
                x2="80"
                y="5"
                y1="5"
                y2="265"
              />
              <line
                fill="none"
                height="260"
                stroke="#ccc"
                stroke-dasharray="3 3"
                width="590"
                x="80"
                x1="670"
                x2="670"
                y="5"
                y1="5"
                y2="265"
              />
            </g>
          </g>
          <g
            class="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
          >
            <line
              class="recharts-cartesian-axis-line"
              fill="none"
              height="260"
              stroke="#666"
              width="60"
              x="20"
              x1="80"
              x2="80"
              y="5"
              y1="5"
              y2="265"
            />
            <g
              class="recharts-cartesian-axis-ticks"
            />
          </g>
        </svg>
        <div
          class="recharts-legend-wrapper"
          style="position: absolute; width: 650px; height: auto; left: 20px; bottom: 5px;"
        />
        <div
          class="recharts-tooltip-wrapper"
          style="transform: translate(undefinedpx, undefinedpx); pointer-events: none; visibility: hidden; position: absolute; top: 0px;"
        >
          <div
            class="recharts-default-tooltip"
            style="margin: 0px; padding: 10px; background-color: rgb(255, 255, 255); border: 1px solid #ccc; white-space: nowrap;"
          >
            <p
              class="recharts-tooltip-label"
              style="margin: 0px;"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
